ingestion/src/kvkafka.rs

```diff
@@ -179,15 +183,128 @@ fn parse_message(message: &str) -> Result<Vec<RawDatum>, Error> {
     Ok(data)
 }
 
+async fn create_timeseries(
+    conn: &mut PooledPgConn<'_>,
+    raw_datum: &RawDatum,
+    permit: Option<i32>,
+) -> Result<i64, Error> {
+    let transaction = conn.transaction().await?;
+
+    // TODO: should we re-check for an existing met label, since the first check was outside the
+    // transaction?
```

Comment:
Hmm, since hypothetically a label could have been created in the meantime? There is definetely a potential race condition here... and it might be important since often what happens is a bunch of new data starts coming in at once (say several elements for one station) and several timeseries have to be created. The worst case is that someone starts trying to dump a new timeseries, but with a bunch of old data... so each piece of data needs the timeseries to exist.  We almost need a lock of some sort on creation of timeseries?

Reply:
Ok, good to explain that it provides the lock. I had not understood that, but I need to go more deeply into the code... so far just a superficial pass.

Reply:
This is fixed now, with a comment explaining the rationale behind the lock